central/graphql/resolvers/image_components_test.go

```diff
@@ -0,0 +1,434 @@
+		return loaders.NewComponentLoader(s.resolver.ImageComponentDataStore)
+	})
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.ImageCVE{}), func() interface{} {
+		return loaders.NewImageCVELoader(s.resolver.ImageCVEDataStore)
+	})
+	s.ctx = loaders.WithLoaderContext(s.ctx)
+
+	// Add Test Data to DataStores
+	testDeployments := testDeployments()
+	for _, dep := range testDeployments {
+		err = s.resolver.DeploymentDataStore.UpsertDeployment(s.ctx, dep)
+		s.NoError(err)
+	}
+
+	testImages := testImages()
+	for _, image := range testImages {
+		err = s.resolver.ImageDataStore.UpsertImage(s.ctx, image)
+		s.NoError(err)
+	}
+}
+
+func (s *GraphQLImageComponentTestSuite) TearDownSuite() {
+	s.envIsolator.RestoreAll()
+
+	imageCVEPostgres.Destroy(s.ctx, s.db)
+	imagePostgres.Destroy(s.ctx, s.db)
+	imageComponentPostgres.Destroy(s.ctx, s.db)
+	imageCVEEdgePostgres.Destroy(s.ctx, s.db)
+	componentCVEEdgePostgres.Destroy(s.ctx, s.db)
+	deploymentPostgres.Destroy(s.ctx, s.db)
+	pgtest.CloseGormDB(s.T(), s.gormDB)
+	s.db.Close()
+}
+
+func (s *GraphQLImageComponentTestSuite) TestUnauthorizedImageComponentEndpoint() {
+	_, err := s.resolver.ImageComponent(s.ctx, IDQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageComponentTestSuite) TestUnauthorizedImageComponentsEndpoint() {
+	_, err := s.resolver.ImageComponents(s.ctx, PaginatedQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageComponentTestSuite) TestUnauthorizedImageComponentCountEndpoint() {
+	_, err := s.resolver.ImageComponentCount(s.ctx, RawQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageComponentTestSuite) TestImageComponents() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	expectedIDs := []string{"comp1#0.9#", "comp2#1.1#", "comp3#1.0#", "comp4#1.0#"}
+	expectedCount := int32(len(expectedIDs))
+
+	comps, err := s.resolver.ImageComponents(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(expectedCount, int32(len(comps)))
+	idList := getIDList(ctx, comps)
+	s.ElementsMatch(expectedIDs, idList)
+
+	count, err := s.resolver.ImageComponentCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(expectedCount, count)
+}
+
+func (s *GraphQLImageComponentTestSuite) TestImageComponentsScoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	imageCompTests := []struct {
+		name        string
+		id          string
+		expectedIDs []string
+	}{
+		{
+			"sha1",
+			"sha1",
+			[]string{"comp1#0.9#", "comp2#1.1#", "comp3#1.0#"},
+		},
+		{
+			"sha2",
+			"sha2",
+			[]string{"comp1#0.9#", "comp3#1.0#", "comp4#1.0#"},
+		},
+	}
+
+	for _, test := range imageCompTests {
+		s.T().Run(test.name, func(t *testing.T) {
+			image := s.getImageResolver(ctx, test.id)
+			expectedCount := int32(len(test.expectedIDs))
+
+			comps, err := image.ImageComponents(ctx, PaginatedQuery{})
+			s.NoError(err)
```

Comment:
this must be `assert.NoError(t,...)`. You are running independent test cases within a suite. 

Same comment for all such check instances.